### PR TITLE
pkg/object: support chacha20 cipher

### LINKED ISSUE #317: Support for modern stream ciphers like Chacha20
**What would you like to be added**:
Support for modern/lightweight stream ciphers like Chacha20 beside current support of AES.

**Why is this needed**:
ChaCha20 takes a 256-bit key and a 32-bit nonce and then creates a key stream, which is then XORed with the plaintext stream. In software, it is three times faster than AES, and is well suited to lower-powered devices and in real time communications.

Quoted from -> https://medium.com/asecuritysite-when-bob-met-alice/aes-is-great-but-we-need-a-fall-back-meet-chacha-and-poly1305-76ee0ee61895

Also a nice coffee-time basic article https://blog.cloudflare.com/do-the-chacha-better-mobile-performance-with-cryptography/

### PR DESCRIPTION
**Pull request by qingyunha**
close #317

### CODE REVIEW

File: pkg/meta/config.go [hunk 1]
```diff
@@ -59,6 +59,7 @@ type Format struct {
 	Inodes           uint64 `json:",omitempty"`
 	EncryptKey       string `json:",omitempty"`
 	KeyEncrypted     bool   `json:",omitempty"`
+	ChaCha20         bool   `json:",omitempty"`
```
Comment (davies) [inline pkg/meta/config.go:62]: I think we should have `EncryptAlgo = chacha20-rsa`, so we can support other algorithm later.

Comment (qingyunha) [reply pkg/meta/config.go]: done

**Comment by davies** [comment]
@qingyunha Thanks for contributing this, it look good overall, more work needed for compatibility, and config/load/dump. 

Since we are close to 1.0 release, it's better to merge this after 1.0 released.

**Comment by qingyunha** [comment]
Ok. Thank you.

File: pkg/meta/config.go [hunk 2]
```diff
@@ -58,6 +58,7 @@ type Format struct {
 	Capacity         uint64 `json:",omitempty"`
 	Inodes           uint64 `json:",omitempty"`
 	EncryptKey       string `json:",omitempty"`
+	EncryptAglo      string `json:",omitempty"`
```
Comment (SandyXSD) [inline pkg/meta/config.go:61]: Should be `EncryptAlgo`

File: pkg/object/encrypt_test.go [hunk 3]
```diff
@@ -126,7 +126,7 @@ func BenchmarkRSA4096Decrypt(b *testing.B) {
 
 func TestAESGCM(t *testing.T) {
 	kc := NewRSAEncryptor(testkey)
-	dc := NewAESEncryptor(kc)
```
Comment (SandyXSD) [inline pkg/object/encrypt_test.go:129]: There are 2 return values

Comment (SandyXSD) [reply pkg/object/encrypt_test.go:129]: Could you also add tests for ChaCha20?

File: cmd/format.go [hunk 4]
```diff
@@ -392,6 +400,7 @@ func format(c *cli.Context) error {
 			SecretKey:    c.String("secret-key"),
 			SessionToken: c.String("session-token"),
 			EncryptKey:   loadEncrypt(c.String("encrypt-rsa-key")),
+			EncryptAglo:  c.String("encrypt-aglo"),
```
Comment (SandyXSD) [inline cmd/format.go:403]: This flag should be ignored like `encrypt-rsa-key` as handled in L388~L389

**Comment by SandyXSD** [comment]
@qingyunha Hi, I leave a few comments, could you continue to complete this PR?

**Comment by qingyunha** [comment]
@SandyXSD Thank you. Updated, please take a look.

**Review by SandyXSD — APPROVED** [review]

**Pull request merged**